ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestCasesRunner.scala

```diff
@@ -112,27 +116,33 @@ final class LedgerTestCasesRunner(
   ): Future[Either[Result.Failure, Result.Success]] =
     result(start(test, session))
 
-  private def run(completionCallback: Try[Vector[LedgerTestSummary]] => Unit): Unit = {
-
-    val system: ActorSystem = ActorSystem(classOf[LedgerTestCasesRunner].getSimpleName)
-    implicit val materializer: Materializer = Materializer(system)
-    implicit val executionContext: ExecutionContext = materializer.executionContext
+  private def uploadDarsIfRequired(participantSessionManager: ParticipantSessionManager)(
+      implicit ec: ExecutionContext): Future[Unit] =
+    if (skipDarUpload) {
+      Future.successful(logger.info("DAR files upload skipped."))
+    } else {
```

Comment:
I generally prefer the `if` clause, rather than the `else` clause, to do the work.

Perhaps we can rename the variable to "uploadDars" instead of "skipDarUpload" and invert the boolean value?